File: API/Models/OrderAggregate/Order.cs
using System.ComponentModel.DataAnnotations;

namespace API.Models.OrderAggregate
{
    public class Order
    {
        public Guid Id {get;set;}
        public string BuyerId {get;set;}
        [Required]
        public ShippingAddress ShippingAddress {get;set;}
        public DateTime OrderDate {get;set;} = DateTime.UtcNow;
        public List<OrderItem> OrderItems {get;set;}
        public long Subtotal {get;set;}
        public long DeliveryFee {get;set;}
        public OrderStatus OrderStatus {get;set;} = OrderStatus.Pending;
        public long GetTotal()
        {
            return Subtotal + DeliveryFee;
        }
    }
}

File: CLIENT/src/app/models/order.ts
export interface Order {
    id: string
    buyerId: string
    shippingAddress: ShippingAddress
    orderDate: string
    orderItems: OrderItem[]
    subtotal: number
    deliveryFee: number
    orderStatus: string
    total: number
  }
  
  export interface ShippingAddress {
    id: string
    fullName: string
    address1: string
    address2: string
    city: string
    state: string
    zip: string
    country: string
  }
  
  export interface OrderItem {
    productId: string
    name: string
    pictureUrl: string
    price: number
    quantity: number
  }